lib/board.rb

```diff
@@ -1,17 +1,40 @@
 # frozen_string_literal: true
 
 class Board
-  attr_accessor :spaces
+  attr_reader :spaces
 
-  def initialize
-    @spaces = (1..9).to_a
+  COMBOS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6]
+  ].freeze
+
+  def initialize(spaces = (1..9).to_a)
+    @spaces = spaces
+  end
+
+  def invalid?(input)
+    /^([1-9])$/.match(input).nil?
+  end
+
+  def occupied?(index)
+    !spaces[index].is_a? Integer
+  end
```

Comment:
Non-blocking and more for informative purposes since you've just learned OCP, but one thing to keep in mind is that what would happen if empty spaces in your board were represented with a non-integer (`" "` or `nil` for example)? This would violate OCP because we'd then have to modify existing code to accommodate for that change.

Reply:
So would it be more in line with OCP to check whether the space had a value matching either of the players' markers?

Reply:
Yep, because depending on what you classify as an empty space, the existence of a marker at that certain index would indicate that space is occupied